### DISCUSSION TITLE
"Killed: 9" for gcc and related binaries

### DISCUSSION THREAD

**Discussion by Feuermurmel** [Homebrew/discussions#3037] [Everyday usage]
I installed `gcc` but invoking e.g. `gcc-11` or `g++-11` does not work:

```
$ opt/homebrew/bin/gcc-ar-11 
Killed: 9
```

Whats wrong here and/or what can I do to fix it?

GCC was installed like this:

```
$ brew install gcc
```

Homebrew was installed like this:

```
cd ~/opt
mkdir homebrew
curl -L https://github.com/Homebrew/brew/tarball/master | tar xz --strip 1 -C homebrew
eval "$(homebrew/bin/brew shellenv)"
brew update --force --quiet
```

More info:

```
$ brew config
HOMEBREW_VERSION: 3.4.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ffc76654da12fba7d34d078fd9bf681fef65921e
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5a3033f1d40994a56c2b9d5e26c47133c58553f8
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /Users/michi/opt/homebrew
HOMEBREW_REPOSITORY: /Users/michi/opt/homebrew
HOMEBREW_CELLAR: /Users/michi/opt/homebrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_EDITOR: see -rw
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_INSTALL_CLEANUP: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false
```

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Your Homebrew's prefix is not /opt/homebrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/opt/homebrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.
```

I can't install Homebrew in the default `/opt/homebrew` because I use multiple user accounts for private and work use.

**Comment by carlocab** [comment]
What's the output of `brew info gcc`?

**Reply by Feuermurmel** [reply]
```
$ brew info gcc
gcc: stable 11.2.0, HEAD
GNU compiler collection
https://gcc.gnu.org/
/Users/michi/opt/homebrew/Cellar/gcc/11.2.0_3 (1,412 files, 333MB) *
  Built from source on 2022-03-07 at 18:24:24
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/gcc.rb
License: GPL-3.0-or-later with GCC-exception-3.1
==> Dependencies
Required: gmp ✔, isl ✔, libmpc ✔, mpfr ✔, zstd ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 99,448 (30 days), 310,104 (90 days), 1,349,329 (365 days)
install-on-request: 39,455 (30 days), 134,074 (90 days), 609,393 (365 days)
build-error: 278 (30 days)
```

**Reply by carlocab** [reply]
I'd try to do
```
brew reinstall $(brew deps gcc)
```
If that doesn't work, try attaching `gcc-11` to a debugger to get a backtrace. That should hopefully produce something useful.

Beyond that I can't help much -- there are very many things that cause broken builds. I'd strongly suggest reconsidering your decision not to use Homebrew's default prefix.

**Reply by Feuermurmel** [reply]
I'd gladly use the single default prefix! I did so with MacPorts for a long time. But Homebrew only supports installed packages to be accessible by a single user (IIUC), per an official Homebrew developer:

>> I'd expect all data installed by Homebrew to be world-readable and all installed software to be runnable by any user.
>
> That's not how homebrew works. Homebrew is only designed for single user usage.

https://github.com/Homebrew/homebrew-core/issues/92551#issuecomment-1005666560

**Comment by gromgit** [comment]
> I can't install Homebrew in the default `/opt/homebrew` because I use multiple user accounts for private and work use.

I don't see why the latter forces the former. In fact, since you're already segregating daily activities into multiple accounts, you might as well add one more just for Homebrew package management, thus neatly segregating software management from everything else.

**Reply by Feuermurmel** [reply]
I can't. It's not supported per an official Homebrew developer:

> That's not how homebrew works. Homebrew is only designed for single user usage.

https://github.com/Homebrew/homebrew-core/issues/92551#issuecomment-1005666560
